arduino/cores/packagemanager/loader.go

```diff
@@ -374,14 +374,39 @@ func (pm *PackageManager) loadBoards(platform *cores.PlatformRelease) error {
 
 	propertiesByBoard := boardsProperties.FirstLevelOf()
 
-	platform.Menus = propertiesByBoard["menu"]
+	if menus, ok := propertiesByBoard["menu"]; ok {
+		platform.Menus = menus
+	} else {
+		platform.Menus = properties.NewMap()
+	}
 
-	delete(propertiesByBoard, "menu") // TODO: check this one
+	// This is not a board id so we remove it to correctly
+	// set all other boards properties
+	delete(propertiesByBoard, "menu")
 
+	skippedBoards := []string{}
 	for boardID, boardProperties := range propertiesByBoard {
-		boardProperties.Set("_id", boardID) // TODO: What is that for??
-		board := platform.GetOrCreateBoard(boardID)
+		var board *cores.Board
+		for _, key := range boardProperties.FirstLevelKeys() {
+			if key == "menu" && platform.Menus.Size() == 0 {
```

Comment:
The code doesn't match with the intention you state in the PR description:
>Boards that define a custom option like stated above won't be loaded anymore if the related menu.MENU_ID=Text is not defined anywhere in the boards.txt.

What this does is skips loading the boards if the boards.txt doesn't define any `menu` properties. It doesn't check whether there is a `menu.MENU_ID` for the `MENU_ID`s used by the board. So if even a single unrelated `menu.MENU_ID` property is defined in the boards.txt, then the boards using `MENU_ID`s without an associated `menu.MENU_ID` property will be loaded.

Reply:
Yeah, you're right. I'll fix it.

Reply:
Should be fixed now, I also modified the `boards.txt` used in the test to verify the new logic.